src/Stratis.Bitcoin.Features.Consensus/Rules/CommonRules/CoinviewRule.cs

```diff
@@ -79,7 +79,7 @@ public override async Task RunAsync(RuleContext context)
                     {
                         this.CheckInputs(tx, view, index.Height);
 
-                        if (!tx.IsCoinStake)
+                        if (!this.Consensus.IsProofOfStake || !tx.IsCoinStake)
```

Comment:
As @zeptin said I think my preference here is to create an abstract method maybe call it `GetTransactionFee` and pass in the `UnspentOutputSet` and the `Transaction` PoW implementation will just calculate the fee, POS will check the `IsCoinStake` condition and return zero if it is. 

`fees += this.GetTransactionFee(tx, view)`

Reply:
Will do shortly.